### DISCUSSION TITLE
Changing the priors

### DISCUSSION THREAD

**Discussion by gsudre** [cdriveraus/ctsem#40] [Q&A] [answered]
Hi @cdriveraus and ctsem community!

I'm getting several issues with "divergent transitions", and reading a bit more of the STAN documentation I'm wondering if it could be related to my priors. I'm running a bivariate ctstem, and the distribution of one of my manifest variables is far from normal (see attached, scaled version). So, my questions are:

1) I've seen examples in the ctsem docs on how to scale the prior in different ways, but how would I go about specifying a non-normal prior?
2) Do you have an intuition about whether that would help with the error? Or maybe other tests/plots/diagnostics I could run to give me a better clue?
3) The reason why that distribution looks that way is the nature of the data. It represents clinical symptoms, and it can go from 0 to 9, and most of the population has none. So, another way to use it would be as an ordered factor, as 1.5 symptoms wouldn't make sense in real life. But I'm not sure if that makes the modeling task any easier?

Thanks!

<img width="998" alt="hist" src="https://user-images.githubusercontent.com/1731397/211299498-02b22e42-5bdd-442a-ae7a-02da3f9814dc.png">

**Comment by cdriveraus** [comment, marked as answer]
1. For a non-normal prior, you would need to decide on an appropriate way to transform the standard normal distribution to place more weight where appropriate. You can visualize the results of this using plot(myctmodel) , though it is probably easier to work out by playing around with the rnorm and density functions of r, like:

```
x=rnorm(10000)
y=x*exp(x/3)
plot(density(y))

```

2. Priors can help with divergences but I doubt it's the manifest mean prior that is the issue. Usually it's something like trying to estimate a correlation (maybe part of T0VAR or a random effects structure, if you have random effects) with limited data. You might want to look at the $priorcheck section of the summary to see if anything stands out, though this won't check everything and is fairly crude. Try simplifying the model and or tightening priors (particularly scale of random effects if any, or std deviation / correlation parameters)

3.  zero inflated data is pretty tough in this kind of case. You need to think about the generating process behind the data -- is there really some scale that every respondent could be placed on and the answers are just censored, or is there really a true zero, no symptom state, that is particularly relevant. Other approaches will give you more flexibility in handling zero inflated data, but they'll probably limit you in other domains. Treat your inferences carefully, check for consistency with other approaches, etc. Simple transforms might help you, though looks like you might have done that already.
